packages/config/src/Project.ts

```diff
@@ -25,3 +29,73 @@ export function getExpoSDKVersion(
     'MODULE_NOT_FOUND'
   );
 }
+
+export function getNativeBuildVersion(
+  config: Pick<ExpoConfig, 'version'> & {
+    android?: Pick<Android, 'versionCode'> & TempRuntimeVersion;
+    ios?: Pick<IOS, 'buildNumber'> & TempRuntimeVersion;
+  },
+  platform: 'android' | 'ios'
+): string {
+  if (!config.version) {
+    throw new ConfigError('Missing "version" field', 'INVALID_CONFIG');
+  }
+  switch (platform) {
+    case 'ios': {
+      if (!config.ios?.buildNumber) {
+        throw new ConfigError(
```

Comment:
I think buildNumber and versionCode might have default values if they're unset. It might be a nicer UX to just use the default values here, too -- as if we were reading them from the actual native build files (even though we're really not).